### ISSUE TITLE
Media Player unavailable and wrong state change to off

### ISSUE THREAD

**Issue by maxalbani** [RogerSelwyn/Home_Assistant_SkyQ_MediaPlayer#63]
**Describe the bug**  
The media player entity is often not available (even outside the night stand by) and I have recorded some wrong state changes from playing to off.
I have automations with an off state that obviously start for no reason (the decoder was not really turned off)

**To Reproduce**  
N/A

**Screenshots**  
![Schermata 2021-09-22 alle 23 05 27](https://user-images.githubusercontent.com/46787665/134484616-04bba072-3218-4d09-a7fe-7e074647937b.png)

**Component versions**  
1. What version of SkyQ component are you using?
v. 2.6.4
2. What was the last version where the problem was not shown?
v. 2.6.4

**Your configuration**  
How have you setup your devices, by UI or YAML?
by UI

**Your configuration from the options dialogue**  
![Schermata 2021-09-23 alle 11 32 00](https://user-images.githubusercontent.com/46787665/134484915-431d1e57-550d-4cfe-be7f-8440f7fdd4d1.png)

**Output of HA logs**  
![Schermata 2021-09-23 alle 11 32 41](https://user-images.githubusercontent.com/46787665/134485020-ce4fec11-8c3a-47f3-8b1e-4a6675692dc9.png)

**Comment by RogerSelwyn** [comment]
This looks like you have an unstable connection between your HA server and your SkyQ box. When I get home over the week-end, I'll provide you a url you can check when it shows unavailable, to validate it is actually inaccessible.

**Comment by maxalbani** [comment]
> Sembra che tu abbia una connessione instabile tra il tuo server HA e il tuo SkyQ box. Quando torno a casa durante il fine settimana, ti fornirò un URL che puoi controllare quando risulta non disponibile, per verificare che sia effettivamente inaccessibile.

Thanks! This integration is very useful for me because it helps me to synchronize the Harmony system in some cases, but so unfortunately it is unusable.

**Comment by maxalbani** [comment]
@RogerSelwyn This is what is happening now. While watching without ever turning off the decoder

![Schermata 2021-09-25 alle 13 33 09](https://user-images.githubusercontent.com/46787665/134770046-c27add83-39d8-4c0b-80d1-cc5e97cc5cf3.png)

**Comment by RogerSelwyn** [comment]
You can check this url at anytime - http://xxx.xxx.xxx.xxx:9006/as/system/information, where xxx is your IP. If there is no response it will show as unavailable (you could also possibly try using a ping). If it gets a response you can tell if the box is on or off by checking the "activeStandby" attribute. Is it True for off, and False for on.

What you show in your screenshot suggests that you have a connectivity issue? Are you using IP address to configure your SkyQ box, or are you doing it by name?

You could potentially trigger a call to that URL from your automation when it changes state, to see if it is reporting validly, but if there is a name lookup issue you may get varying results anyway.

Also, are there any errors in the logs? 

Finally you reported version 2.6.4 as having the problem, also as being the last version that didn't have the problem. Has it ever worked for you? If so, which version was it that worked?

**Comment by maxalbani** [comment]
Thanks! I will check this url when I detect something strange.
I am using ip address, but the box is connected via wifi, maybe the connection is unstable for this.

No errors in the logs, only the warning W0030M.
I started using the component recently, already in version 2.6.4.

I'll let you know if the url gives me correct values when the component is unavailable or when it "playing -> off" for no reason.
Thanks again!

**Comment by RogerSelwyn** [comment]
Bear in mind if the connection is unstable, then it will take some portion of a section to fire an automation, by which time it may be resolved. That said, I would anticipate that it swapped from Unavailable to Off, or Unavailable to Playing/Paused, rather than cycling round all three. Let me know what you find. My personal setup is a Sky Q and a Mini, both on wired connections (over power line) which has proven pretty stable. I also have about 500 downloads of the module, and no-one else has complained of this issue, so my suspicion is it is network issue, but of course I have been proved wrong before.

**Comment by maxalbani** [comment]
@RogerSelwyn 

I have created a binary sensor rest with the provided URL.
When the media player entity is not available, the sensor also goes offline, so there is no connectivity between the HA server and the box.

I solved by excluding these state changes with the conditions in my automations.

I don't know if it's a bug (it's possible that most people don't have automation with the media player, so they don't even notice), or it's just my network problem.
Thanks for the support! I think this issue can be closed.
